client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,469 @@
+                        header-text-variant="white"
+                        align="center"
+                    >
+                        <b-button
+                            @click="setSharing(actions.share_with, shareWithEmail, share_option.make_public)"
+                            v-if="item.extra.can_change.length > 0"
+                            block
+                            variant="outline-primary"
+                            >Make datasets public</b-button
+                        >
+                        <b-button
+                            @click="
+                                setSharing(actions.share_with, shareWithEmail, share_option.make_accessible_to_shared)
+                            "
+                            v-if="item.extra.can_change.length > 0"
+                            block
+                            variant="outline-primary"
+                            >Share only with {{ shareWithEmail }}</b-button
+                        >
+                        <b-button
+                            @click="setSharing(actions.share_with, shareWithEmail, share_option.no_changes)"
+                            block
+                            variant="outline-primary"
+                            >Share Anyway
+                        </b-button>
+                        <b-button @click="getModel()" block variant="outline-danger">Cancel </b-button>
+                    </b-card>
+                </b-col>
+            </b-row>
+            <!--            </b-card-group>-->
+        </div>
+    </div>
+</template>
+
+<script>
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faCopy, faEdit, faUserPlus, faUserSlash } from "@fortawesome/free-solid-svg-icons";
+import { getAppRoot } from "onload/loadConfig";
+import { getGalaxyInstance } from "app";
+import SlugInput from "components/Common/SlugInput";
+import axios from "axios";
+import Multiselect from "vue-multiselect";
+import { copy } from "utils/clipboard";
+
+Vue.use(BootstrapVue);
+[faCopy, faEdit, faUserPlus, faUserSlash].forEach((icon) => library.add(icon));
+
+const defaultExtra = () => {
+    return {
+        cannot_change: [],
+        can_change: [],
+        can_share: true,
+    };
+};
+export default {
+    components: {
+        FontAwesomeIcon,
+        SlugInput,
+        Multiselect,
+    },
+    props: {
+        id: {
+            type: String,
+            required: true,
+        },
+        plural_name: {
+            type: String,
+            required: true,
+        },
+        model_class: {
+            type: String,
+            required: true,
+        },
+    },
+    data() {
+        const Galaxy = getGalaxyInstance();
+        return {
+            isExposeEmail: Galaxy.config.expose_user_email || Galaxy.user.attributes.is_admin,
```

Comment:
I did it. but I have mixed feeling about it. Instead of one line solution, I had to define and register 2 different components. Is there any other way, where I can just call a function? Anyway, it's fine for now

Reply:
With Vue3's composition API I think we can make this a lot cleaner.  We absolutely do need to move away from ever touching `Galaxy` global object, though, so this is the right next step.

Reply:
is it possible to call it directly from Vuex without defining `mapGetters`?